src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -1789,12 +1779,6 @@ void StatefulWriter::send_heartbeat_nts_(
             return;
         }
     }
-    else
-    {
-        assert(firstSeq <= lastSeq);
-
-        // Check if it has to be sent a GAP with the gaps in the history
-    }
```

Comment:
I think we should keep the assertion, and just remove the comment regarding the GAPs